tools/InterfaceGenerator/generator/generators/PolicyTypes.py

```diff
@@ -0,0 +1,687 @@
+
+        get_first_enum_value_name = lambda enum : enum.elements.values()[0].name
+        enum_required_for_policy = lambda enum : enum.name in required_enums_for_policy and "." not in get_first_enum_value_name(enum)
+
+        # In case if "." is in FunctionID name this is HMI_API function ID and should not be included in Policy  enums
+
+        required_enum_values = [val for val in interface.enums.values()
+                               if enum_required_for_policy(val)]
+
+        self._write_header_with_enums(filename, namespace, destination_dir, required_enum_values)
+        self._write_cc_with_enums(filename, namespace, destination_dir, required_enum_values)
+        if filename == "MOBILE_API":
+            self._write_cc_with_enum_schema_factory(filename, namespace, destination_dir, interface.enums.values())
+
+    def _write_cc_with_enum_schema_factory(self,filename, namespace, destination_dir, required_enum_values):
+        class_name = u"generated_{0}_enum_schema_factory".format(filename)
+        header_file_name = '''policy/policy_table/policy_enum_schema_factory.h'''
+        cc_file_name = u"".join("{0}.cc".format(class_name))
+        if not os.path.exists(destination_dir):
+            os.makedirs(destination_dir)
+        print(os.path.join(destination_dir, cc_file_name))
+        with codecs.open(os.path.join(destination_dir, cc_file_name),
+                         encoding="utf-8",
+                         mode="w") as f_cc:
+            guard = u"_{0}_{1}_CC__".format( class_name.upper(),
+                    unicode(uuid.uuid1().hex.capitalize()))
+            namespace_open, namespace_close = self._namespaces_strings(namespace)
+            includes = '''#include <set>\n'''\
+            '''#include "interfaces/MOBILE_API.h"\n'''\
+            '''#include "smart_objects/enum_schema_item.h"\n'''
+            f_cc.write(self._cc_file_tempalte.substitute(
+                class_name=class_name,
+                header_file=header_file_name,
+                includes = includes,
+                namespace_open=namespace_open,
+                enums_content=self.generate_enums_schema_factory_impl(required_enum_values),
+                namespace_close=namespace_close))
+
+    def _write_header_with_enums(self,filename, namespace, destination_dir, required_enum_values):
+        class_name = u"generated_{0}_policy_types".format(filename)
+        header_file_name = u"".join("{0}.h".format(class_name))
+        if not os.path.exists(destination_dir):
+            os.makedirs(destination_dir)
+        with codecs.open(os.path.join(destination_dir, header_file_name),
+                         encoding="utf-8",
+                         mode="w") as f_h:
+            guard = u"_{0}_{1}_H__".format( class_name.upper(),
+                    unicode(uuid.uuid1().hex.capitalize()))
+            namespace_open, namespace_close = self._namespaces_strings(namespace)
+            f_h.write(self._h_file_tempalte.substitute(
```

Comment:
`_h_file_tempalte` -> `_h_file_template`

Reply:
@jacobkeeler fixed in 76277de2068688672613bec5ac92a1beae5574dd